langroid/language_models/azure_openai.py

```diff
@@ -91,6 +91,8 @@ def __init__(self, config: AzureConfig):
         # when you deployed a model
         if "35-turbo" in self.config.model_name:
             self.config.chat_model = OpenAIChatModel.GPT3_5_TURBO
+        elif self.config.model_name == "gpt-4":
```

Comment:
looks odd - you're picking turbo when model name == 'gpt-4'

Reply:
Yeah, I should have put the reference, you can see GPT-4 (with 1106-preview)
https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability 

So when I want to deploy a model using GPT-4-turbo on Azure, I need to select GPT-4
<img width="594" alt="image" src="https://github.com/langroid/langroid/assets/15859139/3fc55ca9-223e-4489-bc4b-ee0a88e67bf3">

This makes the model name (which is a required parameter in AzureConfig) to become gpt-4
<img width="364" alt="image" src="https://github.com/langroid/langroid/assets/15859139/25407b6e-3768-4be9-abf8-d464a8f22cb0">

While the model name if you use gpt-4 will be `gpt-4-32k`